agent/taskresource/cgroup/control/cgroupv2_controller_linux.go

```diff
@@ -0,0 +1,79 @@
+//go:build linux
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package control
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/cihub/seelog"
+	cgroupsv2 "github.com/containerd/cgroups/v2"
+)
+
+const (
+	defaultCgroupv2Path = "/sys/fs/cgroup"
+	parentCgroupSlice   = "/"
+)
+
+// controlv2 is used to implement the cgroup Control interface
+type controlv2 struct{}
+
+// Create creates a new cgroup based off the spec post validation
+func (c *controlv2) Create(cgroupSpec *Spec) error {
+	// Validate incoming spec
+	err := validateCgroupSpec(cgroupSpec)
+	if err != nil {
+		return fmt.Errorf("cgroup create: failed to validate spec: %w", err)
+	}
+
+	cgroupPath := cgroupSpec.Root
+	seelog.Infof("Creating cgroup cgroupv2root=%s parentSlice=%s cgroupPath=%s", defaultCgroupv2Path, parentCgroupSlice, cgroupPath)
+	_, err = cgroupsv2.NewSystemd(parentCgroupSlice, cgroupPath, -1, cgroupsv2.ToResources(cgroupSpec.Specs))
+	if err != nil {
+		return fmt.Errorf("cgroup create: unable to create v2 manager cgroupPath=%s err=%s", cgroupPath, err)
+	}
+
+	return nil
+}
+
+// Remove is used to delete the cgroup
+func (c *controlv2) Remove(cgroupPath string) error {
+	seelog.Infof("Removing cgroup cgroupv2root=%s parentSlice=%s cgroupPath=%s", defaultCgroupv2Path, parentCgroupSlice, cgroupPath)
+
+	m, err := cgroupsv2.LoadSystemd(parentCgroupSlice, cgroupPath)
+	if err != nil {
+		return err
+	}
+	return m.Delete()
+}
+
+// Exists is used to verify the existence of a cgroup
+func (c *controlv2) Exists(cgroupPath string) bool {
+	fullCgroupPath := filepath.Join(defaultCgroupv2Path, parentCgroupSlice, cgroupPath)
+	seelog.Infof("Checking existence of cgroup cgroupv2root=%s parentSlice=%s cgroupPath=%s fullPath=%s", defaultCgroupv2Path, parentCgroupSlice, cgroupPath, fullCgroupPath)
+
+	if _, err := os.Stat(fullCgroupPath); os.IsNotExist(err) {
```

Comment:
we should still log the error here, there's a chance that the file exists but `os.Stat` will return error anyways